dashboard/app/dsl/script_dsl.rb

```diff
@@ -76,21 +78,43 @@ def level_concept_difficulty(json)
 
   string :video_key_for_next_level
 
+  boolean :active
+
   def assessment(name)
     level(name, {assessment: true})
   end
 
   def level(name, properties = {})
-    @levels << {
+    level = {
       :name => name,
-      :stage => @stage,
       :stage_flex_category => @stage_flex_category,
       :skin => @skin,
       :concepts => @concepts.join(','),
       :level_concept_difficulty => @level_concept_difficulty || {},
       :video_key => @video_key_for_next_level
     }.merge(properties).select{|_, v| v.present? }
     @video_key_for_next_level = nil
+    if @current_scriptlevel
+      @current_scriptlevel[:levels] << level
+      if !@active
+        @current_scriptlevel[:properties][name] = { :active => @active }
+        @active = true
+      end
+    else
+      @scriptlevels << {
+        :stage => @stage,
+        :levels => [level]
+      }
+    end
+  end
+
+  def variants
+    @current_scriptlevel = { :levels => [], :properties => {}, :stage => @stage}
+  end
+
+  def endvariants
+    @scriptlevels << @current_scriptlevel
+    @current_scriptlevel = nil
```

Comment:
As we add more concepts to the `.script` format I think we're going to want to eventually create a richer editor than just text.  I don't have a good sense of when the cutoff point is for switching formats (maybe we've already passed it).

If we do go the block route, we should consider moving some existing flat options to nested options.  E.g. all the [metadata for legacy levels](https://github.com/code-dot-org/code-dot-org/blob/e389815a37c0a2eaefbddc9a00685a197f678bd7/dashboard/config/scripts/20-hour.script#L90-L93):
```ruby
...
skin 'pvz'
concepts 'if_else', 'loop_until'
level_concept_difficulty '{"sequencing":5,"repeat_until_while":4,"conditionals":4}'
level 'blockly:Maze:2_18'

skin 'pvz'
concepts 'if_else', 'loop_until'
level_concept_difficulty '{"sequencing":5,"repeat_until_while":4,"conditionals":5}'
level 'blockly:Maze:2_19'
...
```